app/Main.hs

```diff
@@ -4,25 +4,10 @@ module Main where
 import Web.Spock
 import Web.Spock.Config
 
-import Control.Monad.Trans
-import Data.Monoid
-import Data.IORef
-import qualified Data.Text as T
-
-data MySession = EmptySession
-data MyAppState = DummyAppState (IORef Int)
-
 main :: IO ()
-main =
-    do ref <- newIORef 0
-       spockCfg <- defaultSpockCfg EmptySession PCNoDatabase (DummyAppState ref)
-       runSpock 8080 (spock spockCfg app)
+main = do
+  spockCfg <- defaultSpockCfg () PCNoDatabase ()
+  runSpock 8080 (spock spockCfg app)
 
-app :: SpockM () MySession MyAppState ()
-app =
-    do get root $
-           text "Hello World!"
-       get ("hello" <//> var) $ \name ->
-           do (DummyAppState ref) <- getState
-              visitorNumber <- liftIO $ atomicModifyIORef' ref $ \i -> (i+1, i+1)
-              text ("Hello " <> name <> ", you are visitor number " <> T.pack (show visitorNumber))
+app :: SpockM () () () ()
```

Comment:
SpockM is a parameterized value of 4 types. The first is the database connection, the second the session, the third application state or settings, and the last your return type (since we're doing some form of IO and don't care about the return type we return () - an empty tuple representing nothing). Since I haven't configured a database, session, or app settings, those are all the empty tuple as well.